src/nemos/base_regressor.py

```diff
@@ -22,6 +23,28 @@
 from .typing import DESIGN_INPUT_TYPE, SolverInit, SolverRun, SolverUpdate
 
 
+def strip_metadata(arg_num: Optional[int] = None, kwarg_key: Optional[str] = None):
+    """Strip metadata from arg."""
+    if arg_num is None and kwarg_key is None:
+        raise ValueError("Must specify either arg_num or kwarg_key.")
+
+    def decorator(func):
+        """Strip metadata if available."""
+
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            inp = args[arg_num] if arg_num is not None else kwargs[kwarg_key]
+            self._metadata = {
+                "metadata": inp.metadata if hasattr(inp, "metadata") else None,
```

Comment:
```suggestion
                "metadata": inp._metadata if hasattr(inp, "_metadata") else None,
```

`inp.metadata` will always pull a pandas DataFrame -- this will be more relevant in the future when the underlying metadata is a dictionary. I would pull from `_metadata` instead which will ensure that the dictionary is pulled in the future